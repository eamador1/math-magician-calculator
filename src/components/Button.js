import React from 'react';
import PropTypes from 'prop-types';
import '../styles/calculator.css';

function Button({ children, onClick }) {
  const isOperator = (value) => value === 'รท' || value === '-' || value === '+' || value === 'x' || value === '=';

  return (
    <button
      type="button"
      className={`button-container ${isOperator(children) ? 'operator' : ''}`.trimEnd()}
      onClick={onClick}
    >
      {children}
    </button>

  );
}

Button.propTypes = {
  children: PropTypes.string.isRequired, // Define the 'id' prop type as a required string
  onClick: PropTypes.func.isRequired,
};

export default Button;
