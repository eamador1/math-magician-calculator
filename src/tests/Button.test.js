import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../components/Button';

it('renders additional className if value is an operator', () => {
  const onClick = jest.fn();
  const { getByText } = render(
    <Button onClick={onClick}>+</Button>,
  );

  const button = getByText('+');
  fireEvent.click(button);

  expect(button).toHaveClass('button-container operator');
  expect(onClick).toHaveBeenCalledTimes(1);

  expect(button).toMatchSnapshot();
});

it('renders without additional className if value is not an operator', () => {
  const onClick = jest.fn();
  const { getByText } = render(
    <Button onClick={onClick}>5</Button>,
  );

  const button = getByText('5');
  fireEvent.click(button);

  expect(button).toHaveClass('button-container');
  expect(button).not.toHaveClass('operator');
  expect(onClick).toHaveBeenCalledTimes(1);

  expect(button).toMatchSnapshot();
});
