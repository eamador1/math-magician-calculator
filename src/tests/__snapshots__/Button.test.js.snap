// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders additional className if value is an operator 1`] = `
<button
  class="button-container operator"
  type="button"
>
  +
</button>
`;

exports[`renders without additional className if value is not an operator 1`] = `
<button
  class="button-container"
  type="button"
>
  5
</button>
`;
